trunk/SUAVE/Methods/Aerodynamics/AVL/create_avl_datastructure.py

```diff
@@ -181,123 +181,144 @@ def populate_wing_sections(avl_wing,suave_wing):
             N/A
         """         
 
-        
-        if len(suave_wing.Segments.keys())>0: 
+        if len(suave_wing.Segments.keys())>0:
                 # obtain the geometry for each segment in a loop                                            
                 symm                 = avl_wing.symmetric
                 semispan             = suave_wing.spans.projected*0.5 * (2 - symm)
-                avl_wing.semispan    = semispan
+                avl_wing.semispan    = semispan   
+                root_chord           = suave_wing.chords.root
+                segment_percent_span = 0;   
+                segments             = suave_wing.Segments
+                n_segments           = len(segments.keys())
+                segment_sweeps       = []
                 origin               = []
                 origin.append(suave_wing.origin)
-                root_chord           =  suave_wing.chords.root
-                segment_percent_span = 0;   
-                n_segments           = len(suave_wing.Segments.keys())
-
-                segment_sweeps = []
+                
                 for i_segs in range(n_segments):
                         if (i_segs == n_segments-1):
                                 segment_sweeps.append(0)                                  
-                        else: 
-                                if suave_wing.Segments[i_segs].sweeps.leading_edge > 0:
-                                        segment_sweep       = suave_wing.Segments[i_segs].sweeps.leading_edge
+                        else: # this converts all sweeps defined by the quarter chord to leading edge sweep since AVL needs the start of each wing section
+                                #from the leading edge coordinate and not the quarter chord coordinate
+                                if segments[i_segs].sweeps.leading_edge > 0: 
```

Comment:
I did this because the leading edge argument was set to 0 and not None. I will change it to None and do a better condition statement which is more robust